### DISCUSSION TITLE
Passing "reset_url" to reset password endpoint does not work

### DISCUSSION THREAD

**Discussion by andriusign** [directus/directus#4611] [Archive]
Trying to access path /auth/password/request gives the following error:

Error:
{"errors":[{"message":"Url \"https://example.com/reset-password\" can't be used to reset passwords.","extensions":{"code":"INVALID_PAYLOAD"}}]}

Request headers:
headers:
email: "user@email.com"
reset_url: "https://example.com/reset-password"

I am sure this was working previously. Did this functionality get removed? If so, why?

---

### Project details
Directus Version: 9.0.0-rc.51
Environment: production
OS: Linux 4.15.0-29-generic
Node: 14.16.0

**Comment by rijkvanzanten** [comment]
You have to specify the allowed URLs in `PASSWORD_RESET_URL_ALLOW_LIST`. See https://docs.directus.io/reference/environment-variables/#security.

This was added to ensure strangers couldn't request a password for another account with a fake reset url 👍🏻

**Reply by andriusign** [reply]
> List of URLs that can be used as reset_url in /password/request

How would you put several urls in the environment variable etc.? Does the url need to be absolute or can it be relative?

**Reply by rijkvanzanten** [reply]
You can use [a CSV of values](https://docs.directus.io/reference/environment-variables/#type-casting-and-nesting) in there (and/or [prefix it with `array:`](https://docs.directus.io/reference/environment-variables/#environment-syntax-prefix)) 👍🏻

**Reply by rijkvanzanten** [reply]
> Does the url need to be absolute or can it be relative?

Can be either

**Reply by andriusign** [reply]
I have put these settings to .env file, reload reload Docker containers, but Directus is not picking up variable from environment:
![image](https://user-images.githubusercontent.com/14263959/111916420-81534900-8a83-11eb-890f-9cc7cf83aff2.png)

.env file:
![image](https://user-images.githubusercontent.com/14263959/111916438-992acd00-8a83-11eb-8309-36a86c65879b.png)

**Reply by rijkvanzanten** [reply]
Could this be "just" a docker/env thing? I tried this locally (given; outside of docker) and it seems to work as expected

**Reply by andriusign** [reply]
All good now. Had to rebuild after changing `.env`. Since you have to specify the full path starting with http, otherwise Directus complains, this also works well in .env file.
```
PASSWORD_RESET_URL_ALLOW_LIST="${PUBLIC_WEB_URL}/change-password"
USER_INVITE_URL_ALLOW_LIST="${PUBLIC_WEB_URL}/confirm-registration"
```

${PUBLIC_WEB_URL} is the url of my front-end application.

**Comment by Yakito** [comment]
Hi! is there any way this can be achieved in a Directus Cloud instance since we can't set envs there? Thanks!

**Reply by albsa** [reply]
Did you figure this out?

**Reply by web-sujal** [reply]
Did you solved this???

**Reply by albsa** [reply]
@web-sujal I did but had to step away from Directus Cloud. Ended up running Directus on Railway which was way more flexible.

**Comment by alpha-mo** [comment]
> You have to specify the allowed URLs in `PASSWORD_RESET_URL_ALLOW_LIST`. See https://docs.directus.io/reference/environment-variables/#security.
> 
> This was added to ensure strangers couldn't request a password for another account with a fake reset url 👍🏻

was it not possible to have an example in the Docs? it feels like documenting the "forgot password" is poor.
